types/test.tsx

```diff
@@ -35,13 +36,18 @@ export async function testPureRender() {
 
   // helpers
   const {container, rerender, debug} = page
+  expectType<HTMLElement, typeof container>(container)
   return {container, rerender, debug}
 }
 
 export function testRenderOptions() {
   const container = document.createElement('div')
   const options = {container}
-  render(<div />, options)
+  const {container: returnedContainer} = render(<div />, options)
+  // Unclear why TypeScript infers `HTMLElement` here when the the input `container` is `HTMLDivElement`
```

Comment:
This is a known issue with TypeScript. JSX syntax unfortunately can't be inferred to a specific element type. You'd have to use the `React.createElement` syntax, though it's not recommended in most cases.